packages/clerk-js/src/ui/components/KeylessPrompt/use-revalidate-environment.ts

```diff
@@ -0,0 +1,62 @@
+import { useClerk } from '@clerk/shared/react';
+import { useEffect, useReducer } from 'react';
+
+import type { Clerk } from '../../../core/clerk';
+import type { Environment } from '../../../core/resources';
+import { useEnvironment } from '../../contexts';
+
+/**
+ * Revalidates environment on focus, highly optimized for Keyless mode.
+ * Attention: this is not a generic solution, and should not be used for revalidating environment inside UI components that are end-user facing (e.g. SignIn)
+ */
+function useRevalidateEnvironment() {
```

Comment:
Currently there is not pattern for "updating" the environment context after a component has already mounted. This is a quick hack to get this working for Keyless mode, which is a dev facing UI component.